dashboard/app/models/plc/course_unit.rb

```diff
@@ -50,4 +50,40 @@ def get_top_modules_of_each_type_from_user_selections(learning_module_ids_and_we
 
     selected_learning_modules
   end
+
+  def determine_preferred_learning_modules(user)
+    evaluation_level = script.levels.reverse.find {|level| level.class == LevelGroup}
+
+    level_source = UserLevel.find_by(user: user, level: evaluation_level, script: script).try(:level_source)
+    return [] if level_source.nil?
+
+    responses = JSON.parse(level_source.data)
+    learning_module_weights = Hash.new(0)
+
+    responses.each do |level_id, response|
+      level = EvaluationMulti.find(level_id)
+      selected_answer = level.answers[response['result'].to_i]
```

Comment:
Upon further consideration, this is only really possible if the answers are corrupted. But its worth doing anyway.